cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -74,18 +83,20 @@
 import java.util.jar.Manifest;
 import java.util.zip.ZipException;
 import javax.annotation.Nullable;
+import javax.ws.rs.Path;
 
 /**
  * Inspects a jar file to determine metadata about the artifact.
  */
 public class ArtifactInspector {
   private static final Logger LOG = LoggerFactory.getLogger(ArtifactInspector.class);
+  private static final Gson GSON = new Gson();
   private final CConfiguration cConf;
   private final ArtifactClassLoaderFactory artifactClassLoaderFactory;
   private final ReflectionSchemaGenerator schemaGenerator;
   private final File tempDir;
 
-  ArtifactInspector(CConfiguration cConf, ArtifactClassLoaderFactory artifactClassLoaderFactory, File tempDir) {
+  public ArtifactInspector(CConfiguration cConf, ArtifactClassLoaderFactory artifactClassLoaderFactory, File tempDir) {
```

Comment:
creating instance in PluginService, other options is to move that class to this package, but that might be odd.